src/main/java/com/esri/samples/featurelayers/feature_layer_shapefile/FeatureLayerShapefileSample.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2018 Esri.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.esri.samples.featurelayers.feature_layer_shapefile;
+
+import java.io.File;
+
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Alert;
+import javafx.scene.layout.StackPane;
+import javafx.stage.Stage;
+
+import com.esri.arcgisruntime.data.ShapefileFeatureTable;
+import com.esri.arcgisruntime.layers.FeatureLayer;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.view.MapView;
+
+public class FeatureLayerShapefileSample extends Application {
+
+  private MapView mapView;
+
+  @Override
+  public void start(Stage stage) throws Exception {
+
+    try {
+      // create stack pane and application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Feature Layer Shapefile Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create a map with a basemap
+      ArcGISMap map = new ArcGISMap(Basemap.createStreetsVector());
+
+      // set the map to the map view
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // create a shapefile feature table from the local file
+      File shapefile = new File("./samples-data/auroraCO/Public_Art.shp");
+      ShapefileFeatureTable shapefileFeatureTable = new ShapefileFeatureTable(shapefile.getAbsolutePath());
+
+      // use the shapefile feature table to create a feature layer
+      FeatureLayer featureLayer = new FeatureLayer(shapefileFeatureTable);
+      featureLayer.addDoneLoadingListener(() -> {
+        if (featureLayer.getLoadStatus() == LoadStatus.LOADED) {
+          // zoom to the feature layer's extent
```

Comment:
Would change comment to `// zoom to area that displays all features within layer`. Better explains what full extent does.